builtin/aws/lambda/platform.go

```diff
@@ -269,39 +277,65 @@ func (p *Platform) Deploy(
 			reset = true
 		}
 
+		if *curFunc.Configuration.EphemeralStorage.Size != storage {
+			update.EphemeralStorage = &lambda.EphemeralStorage{
+				Size: aws.Int64(storage),
+			}
+			reset = true
+		}
+
 		if reset {
+			step.Update("Detected a configuration change. Updating Lambda function...")
+
 			update.FunctionName = curFunc.Configuration.FunctionArn
 
 			_, err = lamSvc.UpdateFunctionConfiguration(&update)
 			if err != nil {
 				return nil, errors.Wrapf(err, "unable to update function configuration")
 			}
+
+			step.Update("Updated Lambda configuration.")
 		}
 
-		funcCfg, err := lamSvc.UpdateFunctionCode(&lambda.UpdateFunctionCodeInput{
-			FunctionName:  aws.String(src.App),
-			ImageUri:      aws.String(img.Name()),
-			Architectures: aws.StringSlice([]string{architecture}),
-		})
+		step.Done()
 
-		if err != nil {
-			if aerr, ok := err.(awserr.Error); ok {
-				switch aerr.Code() {
-				case "ValidationException":
-					// likely here if Architectures was invalid
-					if architecture != lambda.ArchitectureX8664 && architecture != lambda.ArchitectureArm64 {
-						return nil, fmt.Errorf("architecture must be either \"x86_64\" or \"arm64\"")
+		// the following update has retry behavior in case the
+		// previous `UpdateFunctionConfiguration` call occurs
+		step = sg.Add("Updating function code...")
+		for i := 0; i < 30; i++ {
+			funcCfg, err := lamSvc.UpdateFunctionCode(&lambda.UpdateFunctionCodeInput{
+				FunctionName:  aws.String(src.App),
+				ImageUri:      aws.String(img.Name()),
+				Architectures: aws.StringSlice([]string{architecture}),
+			})
+
+			if err == nil {
+				funcarn = *funcCfg.FunctionArn
+				step.Update("Updated function code!")
+				break
+			}
+
+			if err != nil {
+
+				if aerr, ok := err.(awserr.Error); ok {
+					switch aerr.Code() {
+					// unrecoverable error
+					case "ValidationException":
+						// likely here if Architectures was invalid
+						if architecture != lambda.ArchitectureX8664 && architecture != lambda.ArchitectureArm64 {
+							return nil, fmt.Errorf("architecture must be either \"x86_64\" or \"arm64\"")
```

Comment:
It doesn't look like we're formatting anything, so I think we could return error messages using the grpc status and codes:

```suggestion
							return nil, status.Error(codes.FailedPrecondition, "architecture must be either \"x86_64\" or \"arm64\"")
```

Reply:
Done! - https://github.com/hashicorp/waypoint/pull/3213/commits/dcc917edcaaeb9fae31db8db2320e9f1aaaba61c